lib/shopify_app/access_scopes/user_strategy.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module AccessScopes
+    class UserStrategy
+      class InvalidInput < StandardError; end
+
+      class << self
+        def update_access_scopes?(user_id: nil, shopify_user_id: nil)
+          return update_access_scopes_for_user_id?(user_id) if user_id
+          return update_access_scopes_for_shopify_user_id?(shopify_user_id) if shopify_user_id
```

Comment:
What if the user passes both, and the user doesn't need to update but the shop does?

Reply:
If both are passed in, it would query it based on the `user_id`.

Reply:
This strategy does not concern itself with shop scope updates. That's `ShopStrategy` and if only a shop needs to update, then [OmniAuth](https://github.com/Shopify/shopify_app/pull/1197/files#diff-f173c9e33b743fa6bec8731bfcfe755501e57aea2a9e98a8e14a3ad9444f2004R53) will be able to determine that.

`ShopAccessScopesVerification` concern also does this check for shop scopes on app load.